models/repo.go

```diff
@@ -1402,7 +1401,14 @@ func CreateRepository(doer, u *User, opts CreateRepoOptions) (_ *Repository, err
 		}
 	}
 
-	return repo, sess.Commit()
+	err = sess.Commit()
+
+	// Add to hook queue for created repo after session commit.
+	if u.IsOrganization() {
```

Comment:
If commit fails repo won't be created

Reply:
Error handling for session commit added.

I do not know why amd64 build is failing after I added this commit.

Reply:
That's arm64, I think that's fail unrelated.

Reply:
I restarted build, it was amd64 fail first, now arm64. I restarted again, hopefully that fixes things.